src/user/service.ts

```diff
@@ -117,3 +118,65 @@ export async function validateUserCredentials(user: User, password: string, log:
   return true;
 }
 
+/**
+ * @description - Given a principal User, uses the principal id to find the UserInfo record for a user.
+ * @param user - The principal User to find the UserInfo record for.
+ * @returns The UserInfo record for the user.
+ * @throws NotFound - If the UserInfo record is not found.
+ */
+export async function findUserInfoByUser(user: User): Promise<UserInfo> {
+
+  const result = await db('user_info')
+    .select()
+    .where({principal_id: user.id})
+    .first();
+
+  if (!result) throw new NotFound(`UserInfo for user "${user.id}" not found.`);
+
+  return recordToModel(user, result);
+}
+
+/**
+ * @description - Given a principal User, uses the principal id to find the Principal user and update the UserInfo record for a user.
+ * @param user - The principal User to update the UserInfo record for.
+ * @param userInfo - new UserInfo object to update the UserInfo record with.
+ * @returns The updated UserInfo record.
+ * @throws BadRequest - If the UserInfo record is not updated.
+ */
+export async function updateUserInfo(user: User, userInfo?: UserInfo): Promise<void> {
+
+  if (!userInfo) return;
+
+  const result = await db('user_info')
+    .where({principal_id: user.id})
+    .update(
+      {
+        name: userInfo.name,
+        locale: userInfo.locale,
+        given_name: userInfo.givenName,
+        family_name: userInfo.familyName,
+        birthdate: userInfo.birthDate,
+        address: userInfo.address ? JSON.stringify(userInfo.address) : null,
+        zoneinfo: userInfo.zoneInfo,
+      }
+    );
```

Comment:
The issue with this function is that it assumes that a record already exists for a user, but this should also handle the case where there was no record yet for a user.

So to do this correctly, this should use `.insert()` instead of `.update()`, and then use `onConflict()` to to turn the `INSERT` into an `UPDATE` automatically